docs/requirements/software-requirements.md

```diff
@@ -0,0 +1,276 @@
+Containers are self-sufficient (without external dependencies) and uniform
+(they look the same on the outside). This makes them decoupled from
+infrastructure specifics (such as the OS used by the infrastructure provider).
+The same container can be run on any Linux distribution, cloud provider
+(including AWS, Azure, Google and DigitalOcean) as well as on a developer’s
+local Mac or Windows machine.
+
+(This corresponds to [point 1](infrastructure-requirements.md#1-the-infrastructure-runs-applications-that-are-packaged-as-oci-containers-must-have-sep) of the [Infrastructure Requirements (IR)](infrastructure-requirements.md))
+
+1. The container images *MUST* be published in CLARIAH's container registry at each release. This can be automated (see
+   [IR7](infrastructure-requirements.md#7-the-infrastructure-automatically-builds-the-application-must-have-auto)).
+2. Containers *SHOULD* be kept as lightweight application containers, not including unnecessary runtime components.
+3. Configuration values (such as database connection strings or API URLs) that may vary between deployments
+   *SHOULD* be parameters to the container. This is implemented through environment variables. The infrastructure in
+   turn configures applications through environment variables (see [IR4](infrastructure-requirements.md#4-the-infrastructure-configures-applications-through-environment-variables-must-have-sep) and also related to point 7).
+4. Containers should output all log information to ``stdout`` so it can be captured by the infrastructure (see IR5 and
+   IR6).
+5. Application data (state) that needs to be persistant between runs *MUST* be stored separate from the container (e.g. in a
+   mounted volume) (See point 3 of the [Infrastructure Requirements (IR)](infrastructure-requirements.md))
+
+### 15. Service developers *SHOULD* provide an initial template when multi-container orchestration is needed
+
+If a complex service consists of multiple interacting containers, the developers *SHOULD* provide an initial template in
+the form of a Docker Compose configuration or a Kubernetes deployment configuration that illustrates how the containers
+are orchestrated to form the application. This *MUST* be maintained in a VCS repository (infrastructure as code
+principle). Infrastructure operators can build on this example to deploy the application.
+
+### 17. Services *MUST* be compatible with CLARIAH's authentication and authorization infrastructure
+
+All services open to end-users and which require some form of user authentication *MUST* be compatible with
+CLARIAH's authentication and authorization infrastructure. That is, they should be able to communicate with CLARIAH's
+[SATOSA](https://github.com/IdentityPython/SATOSA) Authentication Provider. It is *RECOMMENDED* to use OpenID Connect
+for this communication. Instruction can be found [here](https://github.com/CLARIAH/IG-DevOps/tree/main/docs/authentication).
```

Comment:
From https://github.com/CLARIAH/IG-DevOps/blob/main/docs/authentication/authentication_clariah_nl.md:

> all of the identity providers registered at CLARIN (see "Participating Identity Federations" at CLARINs Service Provider Federation). This basically means: all European academic institutions. In addition to this CLARIN-provided collection of identity providers, Beeld en Geluid is also available as an identity provider in discovery.clariah.nl.

So this requirement makes a lot of sense for applications geared towards academic users. But what about applications with other types of users? Case in point: NDE’s [Solid Collection Registration System](https://github.com/netwerk-digitaal-erfgoed/solid-crs). What would be the added value for this type of applications in using SATOSA?

So should we:

- generalise this requirement to be about OIDC-compatibility on the part of the application, so it can be connected to _any_ OIDC IdP?
- or document more clearly how to use CLARIAH auth with non-academic users (I guess BenG is an example of this)?